### DISCUSSION TITLE
LSD2: How to resolve polytomies in ML tree?

### DISCUSSION THREAD

**Discussion by aalonsor15** [iqtree/iqtree2#189] [Q&A] [answered]
Hello,
I recently received help from this community on running a dated, rooted, and constrained phylogenetic tree for moths. I used this code:
iqtree2 -s mysequences.fasta -g myconstrain.newick -o "LNAUT645_14_Prototheora_corvifera_CCDB_22946_G04_Hepialidae" --date age_file.txt -B 1000 -bnni -T AUTO --prefix multidated_tree -redo --date-tip 0

The analysis ran and the result looks promising. Here is the log: 
[multidated_tree.log](https://github.com/iqtree/iqtree2/files/15214170/multidated_tree.log)

However, it seems that the tree I get from this analysis is not fully resolved. There are a lot of polytomies. Is there a way to tell IQtree to give me the best/fully resolved tree, or tell it to forbid/resolve polytomies, or tell it to not collapse branches even though they have no clear optimal placement?

Thanks again!

**Comment by roblanf** [comment]
@aalonsor15, I think that IQ-TREE itself will always output a bifurcating tree, unless you use the `-czb` option described in the command reference (@bqminh can probably confirm!).

What I think is happening here is that LSD2 is collapsing your short branches specifically because they are uninformative (covered here: https://github.com/tothuhien/lsd2). I'm not sure what the impact would be of *forcing* LSD2 to keep those branches - it may do strange things to your dating analysis, perhaps @tothuhien could tell you more on that one.

Assuming that to be the case, there are a few options if you really want a fully-resolved, dated tree:

1. Randomly resolve polytomies but keep the dates as in the output file
2. Resolve polytomies according to the ML tree from IQ-TREE (not the dated tree from LSD2). (Many of these may also be ~randomly resolved if there really is zero information on the existence or otherwise of a branch).
3. (And I know this is not what you asked, but I'll put it here for completeness): accept the uncertainty implied by the polytomies

In my opinion, the option(s) you use depend on your question. For example, if you were doing a comparative analysis where the branching order actually mattered, it might be best to repeat the analysis on a large sample of trees with randomly-resolved polytomies, and integrate this into the uncertainties of the parameters you are interested in. 

Another very good option here would be to move to a Bayesian framework, which naturally allows you to integrate _out_ uncertainty in the tree topology with respect to any parameter(s) you are interested in measuring. BEAST2 would be a great place to do that.

Hope some of that helps,

Rob

**Reply by roblanf** [reply]
And from a practical point of view w.r.t. forcing LSD2 not to make polytomies, perhaps this thread on the google group will help you?

https://groups.google.com/g/iqtree/c/KVJ9Tx7YvRE/m/Lu4Gkpy-AAAJ

**Comment by tothuhien** [comment, marked as answer]
Hi, 
Thanks @roblanf for answering the question. So briefly, if you want of force LSD2 not to make polytomies, for example you want the minimum branch length is 0.001, you can add the option:
_--date-options "-l 0 -u 0.001"_
option `-l 0` prevents to collapse small (non-informative) branches at the beginning. The reason we want to collapse small branches is because they are normally not accurately resolved; so we rather consider them to be unresolved to prevent this inaccuracy to propagate to the dating estimation. And `-u` forces all estimated branches to be >= than the set value.
For the side effect of that, the dating tree may not correspond to the optimal solution. By forcing some branch lengths to be a bit longer, the root age may be pushed back to be a bit older. But if the value you put for -u is very small, and not many branches are affected, then the effect should be negligible. 
Kind regards,
Hien

**Comment by aalonsor15** [comment]
Thank you @roblanf and @tothuhien for your answers! They were both very informative and helped me better understand how to move forward with this.

Just for a bit of background, I am constructing a dated phylogenetic tree of moth species from 19 different families, using only DNA barcoding sequences from 1 gene. I collected all of the moths myself in a rainforest, before and after a hurricane. So, I am not interested in resolving a perfect moth phylogeny. I only need this tree so I can calculate phylogenetic diversity and assess how the hurricane impacted phylogenetic diversity of moths. I am using the iNext3D R package for this.

I ran the tree dating again, this time with the --date-options "-l 0 -u 0.001" code that you recommended, and recalculated all phylogenetic diversity metrics with the new tree. This change did not alter the statistical significance of hurricane impacts at all, so I think either option would work for my purposes since they do not change the implications of my results. In this case, should I use the default LSD2 options and accept the polytomies as inherent uncertainty in my data? I could then just mention that we repeated the analysis with a tree that did not collapse branches close to zero, and that did not change our observed results. I think the answer is yes (I should accept the uncertainty implied by the polytomies, as suggested by @roblanf), but wanted to give a bit more background to get your recommendations. 

Thanks again!

**Reply by roblanf** [reply]
@aalonsor15 yes - in my opinion you should keep the polytomies for the main analysis (because as @tothuhien says, resolving them will tend to bias the dating). Including a secondary analysis to show that it wouldn't have mattered if you resolved them is a nice way to show that your results are robust to this decision either way.
